### PR TITLE
[compiler-rt] reimplements GetMemoryProfile for netbsd.

### PR DESCRIPTION
**Pull request by devnexen**
The actual solution relies on the premise /proc/self/smaps existence. instead relying on native api like freebsd.
fixing fuzzer build too.

### CODE REVIEW

File: compiler-rt/lib/fuzzer/FuzzerUtilLinux.cpp [hunk 1]
```diff
@@ -44,7 +44,7 @@ void SetThreadName(std::thread &thread, const std::string &name) {
 #if LIBFUZZER_LINUX || LIBFUZZER_FREEBSD
   (void)pthread_setname_np(thread.native_handle(), name.c_str());
 #elif LIBFUZZER_NETBSD
-  (void)pthread_set_name_np(thread.native_handle(), "%s", name.c_str());
+  (void)pthread_setname_np(thread.native_handle(), "%s", const_cast<char *>(name.c_str()));
```
Comment (vitalybuka) [inline compiler-rt/lib/fuzzer/FuzzerUtilLinux.cpp:47]: looks unrelated

**Review by vitalybuka — APPROVED** [review]
Please check why presubmit complains.

File: compiler-rt/lib/fuzzer/FuzzerUtilLinux.cpp (hunk 1 above)
Comment (devnexen) [reply compiler-rt/lib/fuzzer/FuzzerUtilLinux.cpp:47]: it is, but it does not build w/o it.
